### PR TITLE
shipit-static-analysis: Remove "enabled checks" string detection in clang tidy, refs #1166

### PR DESCRIPTION
**Pull request by La0**
The string "Enabled checks:" can move in the mach static-analysis command output, so we cannot rely anymore on it, otherwise we miss some issues.

### CODE REVIEW

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py [hunk 1]
```diff
@@ -120,16 +120,14 @@ def parse_issues(self, clang_output, revision):
...
-        if end is not None:
-            clang_output = clang_output[:end.start()-1]
-
         # Sort headers by positions
         headers = sorted(
             REGEX_HEADER.finditer(clang_output),
             key=lambda h: h.start()
         )
+        if not headers:
+            logger.warn('No clang-tidy header was found even though a clang output was provided')
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: If this is not supposed to happen, maybe throw, or make the log more severe?

Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: I'll leave it to `warn` as it can happen, when there are genuinely no issues described. This should be a concern only if we see this warning on every report though.

Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: I found a line at the end of the output `XX warnings present.` where `XX` is an integer (itc an be 0, for valid clang files without issues).
This line is not present when no analysis is made (non-clang files), so we can raise an Exception is no header is found when `XX` is present and positive

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py [hunk 2]
```diff
@@ -119,17 +120,25 @@ def parse_issues(self, clang_output, revision):
...
         '''
-
-        # Limit clang output parsing to 'Enabled checks:'
-        end = re.search(r'^Enabled checks:\n', clang_output, re.MULTILINE)
-        if end is not None:
-            clang_output = clang_output[:end.start()-1]
+        # Detect end of file warnings count
+        # When an invalid file is used, this line does not appear
+        has_warnings = REGEX_HAS_WARNINGS.search(clang_output)
+        if has_warnings is None:
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Could you add tests for these two new cases? (empty output and no issues found)

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py (hunk 1 above)
Comment (marco-c) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Sounds good!

File: src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py (hunk 2 above)
Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Not really, because i use clang-tidy directly in the unit test, as `mach` is not available .
`clang-tidy` does not have the line in its output, i append it in `conftest.py`

We **really** need integration tests.

Comment (marco-c) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: You could write a unit test that calls `parse_issues` directly, with a "fake" clang_output based on what you've seen with mach.

Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: Jus to trigger the exception (or the analysis) ? Sure

Comment (marco-c) [reply src/shipit_static_analysis/shipit_static_analysis/clang/tidy.py]: To trigger the exception, and to test these two cases (no output, 0 issues)
